test/integration/Cover/buyCover.js

```diff
@@ -0,0 +1,459 @@
+
+const REWARD_DENOMINATOR = 10000n;
+const PRICE_CHANGE_PER_DAY = 200n;
+
+function calculateRewards(premium, timestamp, period, rewardRatio, bucketDuration) {
+  const expirationBucket = divCeil(BigInt(timestamp) + BigInt(period), bucketDuration);
+  const rewardStreamPeriod = expirationBucket * bucketDuration - BigInt(timestamp);
+  const rewardPerSecond = (premium * rewardRatio) / REWARD_DENOMINATOR / rewardStreamPeriod;
+  return rewardPerSecond * rewardStreamPeriod;
+}
+
+const stakedProductParamTemplate = {
+  productId: 0, // Use Product 0 which allows all pools
+  recalculateEffectiveWeight: true,
+  setTargetWeight: true,
+  targetWeight: 100,
+  setTargetPrice: true,
+  targetPrice: 100,
+};
+
+const buyCoverFixture = (overrides = {}) => ({
+  coverId: 0,
+  owner: ethers.ZeroAddress,
+  productId: stakedProductParamTemplate.productId,
+  coverAsset: PoolAsset.ETH,
+  amount: ethers.parseEther('1'),
+  period: daysToSeconds(30),
+  maxPremiumInAsset: ethers.MaxUint256,
+  paymentAsset: PoolAsset.ETH,
+  commissionRatio: 0,
+  commissionDestination: ethers.ZeroAddress,
+  ipfsData: 'ipfs data',
+  ...overrides,
+});
+
+async function buyCoverSetup() {
+  const fixture = await loadFixture(setup);
+  const { products } = fixture;
+  const [manager1, manager2, manager3] = fixture.accounts.stakingPoolManagers;
+  const { stakingProducts } = fixture.contracts;
+
+  const { targetPrice } = stakedProductParamTemplate;
+  const productIdWithBumpedPrice = products.findIndex(
+    p => targetPrice !== p.product.initialPriceRatio && !p.product.useFixedPrice,
+  );
+  const productIdWithFixedPrice = products.findIndex(
+    p => targetPrice !== p.product.initialPriceRatio && p.product.useFixedPrice,
+  );
+  const productIdIsDeprecated = products.findIndex(p => p.product.isDeprecated);
+
+  await stakingProducts
+    .connect(manager1)
+    .setProducts(1, [
+      stakedProductParamTemplate,
+      { ...stakedProductParamTemplate, productId: productIdWithBumpedPrice },
+      { ...stakedProductParamTemplate, productId: productIdWithFixedPrice },
+      { ...stakedProductParamTemplate, productId: productIdIsDeprecated },
+    ]);
+  await stakingProducts.connect(manager2).setProducts(2, [stakedProductParamTemplate]);
+  await stakingProducts.connect(manager3).setProducts(3, [stakedProductParamTemplate]);
+
+  return fixture;
+}
+
+describe('buyCover', function () {
+  it('allows to buy against multiple staking pools', async function () {
+    const fixture = await loadFixture(buyCoverSetup);
+    const { cover, tokenController, stakingProducts, ramm, pool } = fixture.contracts;
+    const [coverBuyer, coverReceiver] = fixture.accounts.members;
+    const { GLOBAL_REWARDS_RATIO, NXM_PER_ALLOCATION_UNIT, BUCKET_DURATION } = fixture.config;
+    const { productId, period, amount } = buyCoverFixture();
+
+    const product = await stakingProducts.getProduct(1, productId);
+
+    const latestBlock = await ethers.provider.getBlock('latest');
```

Comment:
nit: use `time.latest()` from @nomicfoundation/hardhat-network-helpers here and for the other tests